### DISCUSSION TITLE
API Call for Img2Img workflow

### DISCUSSION THREAD

**Discussion by ckao10301** [Comfy-Org/ComfyUI#1968] [Q&A]
Hi, I'm trying to build an api call for img2img workflow. How would I send the image file to "LoadImage"? I saw that the input image needs to exist in /comfyui/input. How do I write the api call that sends and saves the image in that folder?

**Comment by devingfx** [comment]
You need to upload the image/mask files prior to using it in a LoadImage node...

There are 2 API endpoints for that:
POST /upload/image
https://github.com/comfyanonymous/ComfyUI/blob/b643eae08b7f0c8eb69b77bd61e31009bfb325b9/server.py#L257

POST /upload/mask
https://github.com/comfyanonymous/ComfyUI/blob/b643eae08b7f0c8eb69b77bd61e31009bfb325b9/server.py#L263

Here is how this is done in JS in the browser :
[widgets.ts<br>Line 562 to 569](https://github.com/Comfy-Org/ComfyUI_frontend/blob/ff2d160230e4082b00563f2dbe3c47531bb9645d/src/scripts/widgets.ts#L562-L569)
```js
 async function uploadFile(file, updateNode, pasted = false) {
    // Wrap file in formdata so it includes filename
    const body = new FormData()
    body.append('image', file)
    if (pasted) body.append('subfolder', 'pasted')
    const resp = await api.fetchApi('/upload/image', {
        method: 'POST',
        body
    })
    // ...
}
// ...
await uploadFile(fileInput.files[0], true)
```

I don't know how to upload/POST a multipart/formdata in python :/ but this should be the usual way...
Maybe someone else can help on this !?

**Reply by devingfx** [reply]
You need to wait the endpoint's response to get 1) the completion 2) the real filename on the server (because the is some renaming if a filename already exists...)

**Reply by devingfx** [reply]
Looks like some people have some python code to upload an image:

https://github.com/fidecastro/ComfyUI-serverless/blob/c50ba62e41e63e303557a21804e0330b0fcb048d/comfy_serverless.py#L160-L173
